tools/autograd/templates/Functions.cpp

```diff
@@ -1886,9 +1886,9 @@ std::tuple<Tensor, Tensor, Tensor> batchnorm_double_backward(
   }
   // for half inputs, save_mean, save_invstd are float (ideally, we would cast
   // everything else, but not now)
-  auto mu = unsqueeze_dim1(training ? save_mean.to(input.dtype()) : running_mean, input);
+  auto mu = unsqueeze_dim1(training ? save_mean.to(input) : running_mean, input);
```

Comment:
shouldn't this be `.to(input.scalar_type())` instead?

Reply:
Don't we expect these all to be on the same device anyways? I would prefer to not use scalar_type() if we don't need to.